### PR TITLE
[font] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1555

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `Font` API - https://docs.expo.io/versions/latest/sdk/font - with minor tweaks and additions to the package source code.

Using auto generated docs exposed the following parts of the API, which previously were not documented (on the website):
* `processFontFamily()`
* `unloadAllAsync()`
* `unloadAsync()`
* `UnloadFontOptions`

The `InlineSnack` examples have been moved from methods, to the Usage section located before the API portion, to retain the usual order and structure of the API docs pages.

Within this PR I have also added the `Hooks` section to the generated API docs, as seen on other, manually written, docs pages. 

# Test Plan

The changes and generated content have been tests running docs website on localhost.

# Preview

Screenshot 2021-07-16 at 12 12 09

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-font/src/Font.ts [hunk 1]
```diff
@@ -12,15 +12,16 @@ import {
 const loaded: { [name: string]: boolean } = {};
 const loadPromises: { [name: string]: Promise<void> } = {};
 
+// @needsAudit
 /**
  * Used to transform font family names to the scoped name. This does not need to
  * be called in standalone or bare apps but it will return unscoped font family
  * names if it is called in those contexts.
  * note(brentvatne): at some point we may want to warn if this is called
  * outside of a managed app.
```
Comment (tsapeta) [inline packages/expo-font/src/Font.ts:21]: Perhaps we don't want this to be included in the docs 😂

**Review by tsapeta — APPROVED** [review]
Looks good to me, but let's have a second pair of eyes to check it 👀

File: docs/components/plugins/APISection.tsx [hunk 2]
```diff
@@ -96,6 +99,7 @@ const renderAPI = (
       <>
         <APISectionComponents data={components} componentsProps={componentsProps} />
         <APISectionConstants data={constants} apiName={apiName} />
+        <APISectionMethods data={hooks} header="Hooks" />
```
Comment (cruzach) [inline docs/components/plugins/APISection.tsx:102]: 👍

**Review by cruzach — APPROVED** [review]

File: packages/expo-font/src/Font.ts (hunk 1 above)
Comment (Simek) [reply packages/expo-font/src/Font.ts]: In the latest commit I have moved the note outside of the method description, so it should be no longer visible in the docs.

**Review by tsapeta — CHANGES_REQUESTED** [review]
Just fix the conflict and I'll merge it then 😉

**Pull request merged**